### PR TITLE
feat: [LM-1528] textfield select by searching

### PR DESCRIPTION
**Pull request by lapapooh**
**PR description:**

**Implemented:**

**Checklist:**

- [x] I ran this code locally
- [ ] I wrote the necessary tests
- [ ] My code follows the [style guidelines](http://bit.ly/sd-web-style-guide)
- [ ] I followed the [instructions](http://bit.ly/sd-web-pr) to create a pull request

**JIRA card:**

**Should know about:**

### CODE REVIEW

**Review by umrzoq-toshkentov — APPROVED** [review]

File: packages/__docs__/package-lock.json
Comment (doniyor2109) [inline packages/__docs__/package-lock.json:1]: Please remove it

File: packages/forms/src/text/FormikTextField.tsx [hunk 1]
```diff
@@ -78,6 +114,7 @@ export const FormikTextField: ForwardRefExoticComponent<FormikTextFieldProps> =
           helperText={errorText || helperText}
           disabled={disabled ?? isSubmitting}
           value={format(field.value)}
+          onKeyDown={handleKeyDown}
```
Comment (doniyor2109) [inline packages/forms/src/text/FormikTextField.tsx:117]: Lets respect user's onKeyDown prop. Please also check for defailtPrevented

File: packages/forms/src/text/FormikTextField.tsx [hunk 2]
```diff
@@ -67,6 +71,38 @@ export const FormikTextField: ForwardRefExoticComponent<FormikTextFieldProps> =
...
+              child,
+            ): child is ReactElement<{
+              value: unknown;
+              children?: ReactNode;
+            }> => isValidElement(child),
+          )
+          .map((child) => ({
+            value: child.props.value,
+            label: child.props.children?.toString() || '',
+          }));
```
Comment (doniyor2109) [inline packages/forms/src/text/FormikTextField.tsx:90]: lets extract to function

File: packages/forms/src/text/FormikTextField.tsx [hunk 3]
```diff
@@ -67,6 +71,38 @@ export const FormikTextField: ForwardRefExoticComponent<FormikTextFieldProps> =
...
+
+        const input = event.key.toLowerCase();
```
Comment (doniyor2109) [inline packages/forms/src/text/FormikTextField.tsx:92]: lets ignore meta keys

File: packages/forms/src/text/FormikTextField.tsx [hunk 4]
```diff
@@ -67,6 +71,38 @@ export const FormikTextField: ForwardRefExoticComponent<FormikTextFieldProps> =
...
+
+        if (input) {
+          // Find the first matching option
+          const matchingOption = options.find((option) =>
+            option.label.toLowerCase().startsWith(input),
+          );
+
+          if (matchingOption) {
+            setValue(matchingOption.value);
```
Comment (doniyor2109) [inline packages/forms/src/text/FormikTextField.tsx:101]: lets use parse here

Comment (doniyor2109) [reply packages/forms/src/text/FormikTextField.tsx:101]: Lets call onChange here

File: packages/forms/src/text/FormikTextField.tsx
Comment (doniyor2109) [inline packages/forms/src/text/FormikTextField.tsx:1]: Lets add controller for this change by prop

File: packages/forms/src/text/utils.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+export function getOptionsFromChildren(
+  children: ReactNode,
+): Array<{ value: unknown; label: string }> {
+  // TextField doesn't support Fragment
+  if (isFragment(children)) {
+    return [];
+  }
+
+  const optionsRaw = Children.toArray(children);
+  return optionsRaw
```
Comment (doniyor2109) [inline packages/forms/src/text/utils.ts:18]:
```suggestion
  return Children.toArray(children)
```

**Review by doniyor2109 — APPROVED** [review]

**Pull request merged**